### PR TITLE
AB#39196 connect to mock api

### PR DESCRIPTION
**Pull request by lhao03**
To test:
- TeamID:  b126aba2-a1fb-4e03-acc3-410b5d5af2dc
- ChannelID: 19:DoKOxALlRc5rtcbZdIjW4t4s9RpHU_HdLk037dgQ5zg1@thread.tacv2
- run this query https://graph.microsoft.com/beta/teams/b126aba2-a1fb-4e03-acc3-410b5d5af2dc/channels/19:DoKOxALlRc5rtcbZdIjW4t4s9RpHU_HdLk037dgQ5zg1@thread.tacv2

You can also try creating a channel. 

Known issues:
- request body and response body are ugly. I am working on finding how to make it `codified` and easy copy/paste like it is in web based Graph Explorer.

My own notes:
```
        // b126aba2-a1fb-4e03-acc3-410b5d5af2dc
        // 19:DoKOxALlRc5rtcbZdIjW4t4s9RpHU_HdLk037dgQ5zg1@thread.tacv2
        // https://graph.microsoft.com/v1.0/teams/b126aba2-a1fb-4e03-acc3-410b5d5af2dc/channels/19:DoKOxALlRc5rtcbZdIjW4t4s9RpHU_HdLk037dgQ5zg1@thread.tacv2
        // /teams/b126aba2-a1fb-4e03-acc3-410b5d5af2dc/channels
        // {
        //     "displayName": "Architecture Discusswerwion",
        //         "description": "This channel is where we debate all future architecture plans"
        // }
```

I plan on refactoring in another PR, because the entire code base needs cleaning up: https://github.com/LokiLabs/microsoft-graph-explorer-teams-app/pull/33#discussion_r680187086

### CODE REVIEW

File: tabs/src/components/sample/QueryRunner.jsx [hunk 1]
```diff
@@ -67,21 +68,37 @@ export function QueryRunner() {
...
     };
 
-    const callGraph = () => {
-        // stub
+    async function callGraph() {
+        // c7ced617-91ba-4d83-8f69-77c39c6eee68
+        // 19:c213c3137b404a6a83e75a5d5142acaf@thread.tacv2
+        // https://graph.microsoft.com/v1.0/teams/c7ced617-91ba-4d83-8f69-77c39c6eee68/channels/19:c213c3137b404a6a83e75a5d5142acaf@thread.tacv2
+        const queryParameters = query.substring(GRAPH_URL.length + graphVersion.length, query.length);
+        const DEVX_API = "https://graphwebapi.azurewebsites.net/graphproxy/" + graphVersion + queryParameters;
```
Comment (omersayshi) [inline tabs/src/components/sample/QueryRunner.jsx]: I know this is still in draft, but its good if we have a constants file that we keep the URL link in

File: tabs/src/components/sample/QueryRunner.jsx [hunk 2]
```diff
@@ -66,25 +68,62 @@ export function QueryRunner() {
...
+            graphResponseHeaders.push({
+                key: p[0],
+                items: [p[0], p[1]]
+            });
+        }
+        setResponseHeaders(graphResponseHeaders);
+        setReponseState(graphResponse.status + " " + graphResponse.statusText);
+        const text = await graphResponse.json();
+        setResponseBody(JSON.stringify(text, undefined, 4));
+    }
```
Comment (omersayshi) [inline tabs/src/components/sample/QueryRunner.jsx]: In our cleanup stories, we should have tasks to create like a 'utilities' folder where we have actions like these in their own independent folder for easier readability and having a more function based repo

Comment (lhao03) [reply tabs/src/components/sample/QueryRunner.jsx]: Yeah, I'll do that in my code refactor story!

File: tabs/src/components/sample/QueryRunner.jsx [hunk 3]
```diff
@@ -67,25 +69,62 @@ export function QueryRunner() {
...
+                headers: cleanedHeaders,
+            };
+        } else {
+            requestParams = {
+                method: requestType,
+                headers: cleanedHeaders,
+                body: requestBody
+            };
+        }
+        const graphResponse = await fetch(url, requestParams);
```
Comment (omersayshi) [inline tabs/src/components/sample/QueryRunner.jsx]:
```suggestion
        let options = {
            method: requestType,
            headers: cleanedHeaders,
        };

        if (requestType !== requestTypes.GET) {
            options.body = requestBody;
        }

        const graphResponse = await fetch(url, options);
```

- Its cleaner if you assign a new key value with the condition that its not a GET call
- Nit: Its better to call it option, to stay consistent with how fetch is usually used

File: tabs/src/components/sample/QueryRunner.jsx [hunk 4]
```diff
@@ -67,25 +69,62 @@ export function QueryRunner() {
...
-        setResponseBody("");
-        setResponseHeaders([{ key: 'Content', items: ['Content', 'json'] }]);
-    };
+    async function callGraph() {
+        const queryParameters = query.substring(GRAPH_URL.length + graphVersion.length, query.length);
+        const url = RSC_API_URL + graphVersion + queryParameters;
+        const cleanedHeaders = {};
+        for (const i of requestHeaders) {
+            cleanedHeaders[i.items[0]] = i.items[1];
+        }
```
Comment (omersayshi) [inline tabs/src/components/sample/QueryRunner.jsx]:
```suggestion
        for (const header of requestHeaders) {
            cleanedHeaders[header.items[0]] = header.items[1];
        }
```

nit: for easier readability

**Review by ayubun — DISMISSED** [review]
I actually don't think any of the changes I'm about to mention are necessary for this specific PR, but I do think they should exist before we consider this "done"; I was wondering if it would be possible to clear the old content out if the request fails? Also, to show a loading icon when the request begins, similar to the main GE? Otherwise, it is actually impossible to tell if a query is being run :eyes:

**Comment by lhao03** [comment]
> I actually don't think any of the changes I'm about to mention are necessary for this specific PR, but I do think they should exist before we consider this "done"; I was wondering if it would be possible to clear the old content out if the request fails? Also, to show a loading icon when the request begins, similar to the main GE? Otherwise, it is actually impossible to tell if a query is being run 👀

When it fails it returns the failed response body right now: 
image

You would want the response body to return just like blank?

Loading icon is also good idea, but I wasn't sure how to display it (where on the app). Any ideas?

**Comment by anneguo3** [comment]
> > I actually don't think any of the changes I'm about to mention are necessary for this specific PR, but I do think they should exist before we consider this "done"; I was wondering if it would be possible to clear the old content out if the request fails? Also, to show a loading icon when the request begins, similar to the main GE? Otherwise, it is actually impossible to tell if a query is being run 👀
> 
> When it fails it returns the failed response body right now:
> image
> 
> You would want the response body to return just like blank?
> 
> Loading icon is also good idea, but I wasn't sure how to display it (where on the app). Any ideas?

Let's open a new issue for the loading issue - and for behaviour the expectation can be to have a "loading" state on the Run query button similar to the how the button loads in the configuration panel when you first add the app to a given chat or channel

File: tabs/src/components/sample/QueryRunner.jsx (hunk 3 above)
Comment (lhao03) [reply tabs/src/components/sample/QueryRunner.jsx]: ac5e50a6af607d75f2f13ce91cea9eedd89751ea

File: tabs/src/components/sample/QueryRunner.jsx (hunk 4 above)
Comment (lhao03) [reply tabs/src/components/sample/QueryRunner.jsx]: ac5e50a6af607d75f2f13ce91cea9eedd89751ea

**Review by jerryxihe — APPROVED** [review]

**Review by omersayshi — APPROVED** [review]
Looks good for now. We should create a task to only allow the current resource to be called and passing the tenant id along.

**Review by ayubun — APPROVED** [review]

**Pull request merged**